### PR TITLE
Remove unnecessary code related to scanning

### PR DESCRIPTION
**Pull request by dbjorge**
#### Description of changes

This PR removes most of the dead code related to scanning and custom rule configurations. In particular:

* We never customize a rule's helpUrl, so the paths in the ResultsDecorator related to injecting custom ones were removed. This incidentally fixes a bug where we were "helpfully" overriding axe's perfectly good helpUrls even if we didn't have a custom one to use (ie, we replaced all the axe help urls with empty strings every time). Fixing this bug was the only change to the snapshotted test output.
* We never customized a rule's help or description, so the paths in the MessageDecorator related to them were removed
* We never customized a check's pass/fail message, so all paths through the check-message-transformer and the associated logic in MessageDecorator were removed
* The only other path in the MessageDecorator was custom decorateNode methods. We did have a single rule with a custom decorateNode (link-functions), but all it did was populate a property the rest of the converter never read, so I removed that path and the MessageDecorator with it
* At that point, none of the rule customizations were used, so I removed all of them.
* Removed all properties of ScannerOptions not used by the converter (and renamed to ConverterOptions)
* Removed a few helpers (role-utils, document-utils) that became dead code after the above

#### Pull request checklist

- [x] Addresses an existing issue: Partial impl of user story 1492620
- [x] Added relevant unit test for your changes. (`npm run test`)
- [x] Verified code coverage for the changes made.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,37 +1,27 @@
...
 import { SarifConverter } from './sarif-converter';
 import { SarifLog } from './sarif/sarifLog';
 import { rulesWCAGConfiguration } from './wcag-mappings';
 
-export function axeToSarif(axeResults: Axe.AxeResults): SarifLog {
-    const messageDecorator = new MessageDecorator(
-        CustomRuleConfigurations,
-        new CheckMessageTransformer(),
-    );
+export { ConverterOptions } from './converter-options';
```
Comment (smoralesd) [inline src/index.ts]: `ConverterOptions` seems duplicated from line 4

File: src/index.ts [hunk 2]
```diff
@@ -1,37 +1,27 @@
...
+    options?: ConverterOptions,
+): SarifLog {
+    options = options ? options : {};
 
-    resultDecorator.setWCAGConfiguration(rulesWCAGConfiguration);
+    const resultDecorator = new ResultDecorator(rulesWCAGConfiguration);
 
     const sarifConverter = new SarifConverter();
 
     // AxeResults -> ScannerResults
```
Comment (smoralesd) [inline src/index.ts]: should the comment be updated? Change `SannerResults` to `ConverterResults`? or maybe just remove the comment?

File: src/index.ts (hunk 1 above)
Comment (dbjorge) [reply src/index.ts]: line 10 exports, line 4 imports

File: src/index.ts (hunk 2 above)
Comment (dbjorge) [reply src/index.ts]: The comment is technically still accurate; I changed the name of the ScannerOptions type, not the ScannerResults type.

**Pull request merged**